chamge/amqp-edge-backend.c

```diff
@@ -81,51 +103,47 @@ _amqp_declare_queue (const amqp_connection_state_t conn,
         amqp_queue_declare (conn, 1, amqp_cstring_bytes (queue_name), 0, 0, 0,
         1, amqp_empty_table);
   }
-  if (amqp_r == NULL) {
-    g_error ("Can not declare a new queue or queue does not exist...");
-    return CHAMGE_RETURN_FAIL;
-  }
+
+  RETURN_FAIL_IF_TRUE (amqp_r == NULL, "error in amqp_queue_declare");
   amqp_get_rpc_reply (conn);
 
   g_assert_nonnull (amqp_r->queue.bytes);
   response_queue_name = g_strdup (amqp_r->queue.bytes);
-  *queue = amqp_cstring_bytes (response_queue_name);
+  if (queue != NULL)
+    *queue = amqp_cstring_bytes (g_steal_pointer (&response_queue_name));
 
   g_debug ("create private reply_to queue : name -> %s",
       (gchar *) queue->bytes);
   return CHAMGE_RETURN_OK;
 }
 
-static ChamgeReturn
+static void
 _amqp_property_set (amqp_basic_properties_t * const props,
     const guint correlation_id, const gchar * reply_queue_name)
 {
-  gchar *correlation_id_str = NULL;
-  gchar *content_type_str = g_strdup (DEFAULT_CONTENT_TYPE);
+  g_autofree gchar *correlation_id_str = NULL;
+  g_autofree gchar *content_type_str = g_strdup (DEFAULT_CONTENT_TYPE);
 
   props->_flags = AMQP_BASIC_CONTENT_TYPE_FLAG | AMQP_BASIC_DELIVERY_MODE_FLAG;
-  props->content_type = amqp_cstring_bytes (content_type_str);
+  props->content_type =
+      amqp_cstring_bytes (g_steal_pointer (&content_type_str));
```

Comment:
If `props->content_type` is already existed, it can be leaked.
Before assigning new string, it should be freed.

In addition, if `content_type_str` is not used in the rest of this block, this can be replaced with;
```
amqp_cstring_bytes (g_strdup (DEFAULT_CONTENT_TYPE));
```